tests/test_lightningd.py

```diff
@@ -643,9 +654,9 @@ def test_connect_by_gossip(self):
         l2 = self.node_factory.get_node()
         # Force l3 to give its address.
         l3port = self.node_factory.get_next_port()
-        l3 = self.node_factory.get_node(options={"ipaddr": "127.0.0.1:{}".format(l3port)})
+        l3 = self.node_factory.get_node(options={"addr": "127.0.0.1:{}".format(l3port)})
```

Comment:
Previous comment not yet covered?  This can be replaced outright now since we now use `addr` in the factory, which also adds the address in the announcement now; basically the `addr` specification here is now redundant.

    -        l3port = self.node_factory.get_next_port()
    -        l3 = self.node_factory.get_node(options={"addr": "127.0.0.1:{}".format(l3port)})
    +        l3 = self.node_factory.get_node()

We can also remove the `get_next_port` method in the factory too, since I added that only to be able to get at the port number for `l3`, and paving the way for ephemeral ports.

Reply:
Yes, decided to leave that for the ephemeral port changes.

Reply:
I have a patch that randomizes ports and allows parallel tests, I'll clean up and open a or tomorrow

Reply:
@cdecker, I like the current system where if I have this:

    l1 = self.node_factory.get_node()
    l2 = self.node_factory.get_node()
    l3 = self.node_factory.get_node()

In the logs, I can filter:

    lightningd(16331) -> l1
    lightningd(16332) -> l2
    lightningd(16333) -> l3

So even with randomized ports, I think it would be nice if each test could show the index of the `lightningd` easily in the logs.

One possibility would be to have a node-factory-factory in the `self` object.  This allocates IDs in batches of 100.  Then a node-factory created for each individual test allocates an ID for each created `lightningd`.  So for example one test would have `l1` -> `lightningd(101)`, `l2` -> `lightningd(102)`... and another test would have `l1` -> `lightningd(201)`, `l2` -> `lightningd(202)`...

Reply:
Yep, I have added a `lightningd-{node_id}` prefix that'll allow to filter by a specific node. See commit 3b33a5c65945e0d5b82a81af466188350b3a5329 for details